solar/solar/orchestration/graph.py

```diff
@@ -157,3 +157,21 @@ def report_topo(uid):
             data.get('end_time')])
 
     return report
+
+
+def wait_finish(uid, timeout, interval=3):
+    """Wait finish will periodically load graph and check if there is no
+    PENDING or INPROGRESS
+    """
+    start_time = time.time()
+
+    while start_time + timeout >= time.time():
+        dg = get_graph(uid)
```

Comment:
we should probably think how we could avoid getting whole graph always, but only changed things.